### PR TITLE
fix xcatprobe xcatmn issues 5677&5678

### PR DESCRIPTION
**Pull request by xuweibj**
### The modification include

_##allow chrony for ntp_
If ``chronyc`` is installed on MN/SN, use ``chronyc`` to check whether ntp service is ready.
If not, still use ``ntpq`` to check.

_##allow rsync dir_
Based on doc https://xcat-docs.readthedocs.io/en/stable/advanced/hierarchy/define_service_nodes.html, tftpdir & installdir could be mounted and rsynced.
If ``sharedtftp=1``, check tftpdir whether mounted.
If ``installloc="/install"``, check installdir whether mounted.
If ``sharedtftp`` not 1, check tftpdir on SN whether same with MN's.
If ``installloc=``, check installdir on SN whether same with MN's.

_##allow name of SN contain other character_
grep ``\w\S+`` as node name.

### The UT result
1. If ``ntpd`` is down, but ``chronyd`` is running:
```
....
[mn]: Checking NTP service is configured...                                                                       [ OK ]
[mn]: Checking rsyslog service is configured...                                                                   [ OK ]
[mn]: Checking firewall is disabled...                                                                            [ OK ]
```

2. When ``installloc=``, and there are differences between MN and SN installdir.
```
[c910f03c17-k17]: Checking 'passwd' table is configured...                                                        [ OK ]
[c910f03c17-k17]: Checking important directories(installdir,tftpdir) are configured...                            [WARN]
[c910f03c17-k17]: installdir '/install' is not the same with MN's, may be error
[c910f03c17-k17]: Checking SELinux is disabled...                                                                 [ OK ]
```

### CODE REVIEW

**Comment by robin2008** [comment]
@xuweibj   One PR is only for one issue.

File: xCAT-probe/lib/perl/probe_utils.pm [hunk 1]
```diff
@@ -524,8 +524,29 @@ sub parse_node_range {
...
+        }
+        if ($chronycoutput =~ /Leap status     : (.+)/) {
+            my $status = $1;
+            if ($status eq "Not synchronised") {
+                $$errormsg_ref = "chronyd did not synchronize.";
+                return 0;
+            }
+        }
+        return 1;
+    }
```
Comment (hu-weihua) [inline xCAT-probe/lib/perl/probe_utils.pm]: Could you refine the structure of this function to let it better to read and maintain? thanks

File: xCAT-probe/subcmds/xcatmn [hunk 2]
```diff
@@ -1281,23 +1316,23 @@ while ($hierarchy_instance->read_reply(\%reply_cache)) {
                 #print ">>>$reply_cache{$servers}->[$_]<<<\n";
                 #For cases like below:
                 #c910f02c04p04: [ok]     :All xCAT daemons are running
-                if ($reply_cache{$servers}->[$_] =~ /^(\w+)\s*:\s*(\[\w+\]\s*):\s*(.*)/) {
+                if ($reply_cache{$servers}->[$_] =~ /^(\w\S+)\s*:\s*(\[\w+\]\s*):\s*(.*)/) {
```
Comment (hu-weihua) [inline xCAT-probe/subcmds/xcatmn]: The node name  can not include specific character, adding ``\S+`` is not correct.

**Comment by hu-weihua** [comment]
It is ok for me, thanks @xuweibj

**Pull request merged**